hack/devtools/platform_msi.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+
+# This script creates a platform Identities to use for local development
+# The script reads the env var PLATFORM_WORKLOAD_IDENTITY_ROLE_SETS and creates platform identities for each operator
+
+get_platform_workloadIdentity_role_sets() {
+    local platformWorkloadIdentityRole
+   
+    # Parse the JSON data using jq
+    platformWorkloadIdentityRoles=$(echo "${PLATFORM_WORKLOAD_IDENTITY_ROLE_SETS}" | jq -c '.[].platformWorkloadIdentityRoles[]')
+
+    echo "${platformWorkloadIdentityRoles}"
+}
+
+assign_role_to_platform_identity() {
+    local principalId=$1
+    local roleId=$2
+    local scope="/subscriptions/${AZURE_SUBSCRIPTION_ID}/resourceGroups/${RESOURCEGROUP}"
+
+    echo "INFO: Assigning roles to platform identity: ${principalId}"
+    local result=$(az role assignment create --assignee "${principalId}" --role "${roleId}"  --scope "${scope}" --output json)
+
+    echo "Role assignment result: $result"
+    echo ""
+}
+
+create_platform_identity_and_assign_role() {
+    local operatorName="${1}"
+    local roleDefinitionId="${2}"
+
+    echo "INFO: Creating platform identity for operator: ${operatorName}"
+    local result=$(az identity create --name "${operatorName}" --resource-group "${RESOURCEGROUP}" --subscription "${AZURE_SUBSCRIPTION_ID}" --output json)
+
+    # Extract the client ID, principal Id, resource ID and name from the result
+    clientID=$(echo $result | jq -r .clientId)
+    principalId=$(echo $result | jq -r .principalId)
+    resourceId=$(echo $result | jq -r .id)
+    name=$(echo $result | jq -r .name)
+
+    echo "Client ID: $clientID"
+    echo "Principal ID: $principalId"
+    echo "Resource ID: $resourceId"
+    echo "Name: $name"
+    echo ""
+
+    if [[ "${operatorName}" == "MachineApiOperator" || "${operatorName}" == "NetworkOperator" \
+        || "${operatorName}" == "AzureFilesStorageOperator" || "${operatorName}" == "ServiceOperator" ]]; then
+
+        assign_role_to_platform_identity "${principalId}" "${roleDefinitionId}"
+    fi
+}
+
+setup_platform_identity() {
+    local platformWorkloadIdentityRoles=$(get_platform_workloadIdentity_role_sets)
+
+    echo "INFO: Creating platform identities under RG ($RESOURCEGROUP) and Sub Id ($AZURE_SUBSCRIPTION_ID)"
```

Comment:
Extending the conversation here, curious to know how the created managed identities will be assigned to the cluster creation process? And if we are going to use the same managed identities for all the dev cluster creations, the federated credentials for the managed identities are limited to 10. That is if federated credentials are not deleted the MIWI cluster creation will be limited to 10.

Reply:
My understanding just from past assumptions and by reading this PR is that the identities will be in the network resource group. They won't be re-used in dev but we can get around the limitation by creating additional resource groups (since the identities need to have unique names)

Maybe we should modify the script to name the identities $CLUSTER-$operatorName or similar?

Reply:
fixed